dagster/pandas_kernel/__init__.py

```diff
@@ -97,3 +100,36 @@ def single_path_arg(input_name, path):
 
 def json_input(name):
     return create_json_input(name)
+
+
+# XXX(freiksenet): this should really go to main solid file
+def validate_transform_fn(solid_name, transform_fn, inputs):
+    input_names = set(inp.name for inp in inputs)
+    used_inputs = set()
+    has_kwargs = False
+
+    signature = inspect.signature(transform_fn)
+    for i, param in enumerate(signature.parameters.values()):
+        if param.kind == inspect.Parameter.VAR_KEYWORD:
+            has_kwargs = True
+        elif param.kind == inspect.Parameter.VAR_POSITIONAL:
+            check.failed(
+                f"solid '{solid_name}' transform function has positional vararg parameter '{param.name}'. Transform functions should only have keyword arguments that match input names and optionally a first positional parameter named 'context'."
+            )
+        else:
+            # XXX(freiksenet): I don't like this
+            if i == 0 and param.name == 'context':
+                pass
+            elif param.name not in input_names:
+                check.failed(
+                    f"solid '{solid_name}' transform function has parameter '{param.name}' that is not one of the solid inputs. Transform functions should only have keyword arguments that match input names and optionally a first positional parameter named 'context'."
+                )
+            else:
+                used_inputs.add(param.name)
+
+    undeclared_inputs = input_names - used_inputs
+    if not has_kwargs and undeclared_inputs:
+        undeclared_inputs_printed = ", '".join(undeclared_inputs)
+        check.failed(
```

Comment:
So the other thing here is that this isn't what check is meant to be used for. (I have been inconsistent on this as well I just want to explain my thinking)

- Check generally indicates a internal framework error akin to a hard error in a programming language. They are supposed to be used where asserts would be used, or a typing in error if python had real types.
- When you are explicitly throwing errors that are dagster-user facing, i.e. communicating "expected" errors rather than internal framework errors, the idea is to use something in dagster.core.errors. I'd recommend DagsterInvalidDefinitionError.

Reply:
Makes sense, I'll fix it to be this.

Reply:
This is fixed.